src/kernel-install/kernel-install

```diff
@@ -77,18 +77,14 @@ if [[ -f /etc/machine-id ]]; then
     read MACHINE_ID < /etc/machine-id
 fi
 
-if ! [[ $MACHINE_ID ]]; then
-    echo "Could not determine your machine ID from /etc/machine-id." >&2
-    echo "Please run 'systemd-machine-id-setup' as root. See man:machine-id(5)" >&2
-    exit 1
-fi
-
 if [[ ! $COMMAND ]] || [[ ! $KERNEL_VERSION ]]; then
     echo "Not enough arguments" >&2
     exit 1
 fi
 
-if [[ -d /efi/loader/entries ]] || [[ -d /efi/$MACHINE_ID ]]; then
+if ! [[ $MACHINE_ID ]]; then
+    BOOT_DIR_ABS=$(mktemp -d /tmp/kernel-install.XXXXX) || exit 1
```

Comment:
So this will "leak" the temporary directory on each premature exit, e.g. when `kernel-install add` is called without the proper arguments. Please install the cleanup function in a trap, see https://github.com/systemd/systemd/blob/master/test/hwdb-test.sh#L32.